src/main/java/net/dv8tion/jda/api/entities/Message.java

```diff
@@ -226,6 +226,44 @@
     @Nullable
     Message getReferencedMessage();
 
+    /**
+     * This returns the id of the referenced message, never null
+     *
+     * @throws java.lang.IllegalStateException
+     *         If this message does not have a reference
+     *
+     * @return The referenced message id
+     */
+    long getReferencedMessageIdLong();
+
+    /**
+     * Retrieves the referenced message for this message. If one is already present, it will be returned.
+     *
+     * @throws java.lang.IllegalStateException
+     *         If this message does not have a reference
+     *
+     * @return {@link net.dv8tion.jda.api.requests.RestAction RestAction} - Type: {@link net.dv8tion.jda.api.entities.Message}
+     */
+    RestAction<Message> retrieveReferencedMessage();
+
+    /**
+     * This returns the id of the referenced message, never null
+     *
+     * @throws java.lang.IllegalStateException
+     *         If this message does not have a reference
+     *
+     * @return The referenced message id
+     */
+    @Nonnull
+    String getReferencedMessageId();
+
+    /**
+     * This determines whether a message has a reference, for use with {@link Message#retrieveReferencedMessage()}
+     *
+     * @return Whether this message has a reference attached to it
+     */
+    boolean hasReferencedMessage();
```

Comment:
Wouldn't it be better to simply have a single getter for all of this instead of 5?

```java
@Nullable
MessageReference getMessageReference();
```